include/graylog_logger/LogUtil.hpp

```diff
@@ -73,20 +97,44 @@ struct LogMessage {
   }
 };
 
+/// \brief The base class used to implement log message consumers.
+///
+/// Inherit from this class when implementing your own log message handler.
+/// Implements queuing of log messages.
 class BaseLogHandler {
 public:
+  /// \brief Does minimal set-up of the this class.
+  /// \param[in] maxQueueLength The maximum number of log messages stored.
   BaseLogHandler(const size_t maxQueueLength = 100);
   virtual ~BaseLogHandler() = default;
+  /// \brief Called by the logging library when a new log message is created.
+  /// \note If the queue of messages is full; any new messages are discarded
+  /// without any indication that this has been done.
+  /// \param[in] msg The log message.
   virtual void AddMessage(const LogMessage &msg);
+  /// \brief Are there messages in the queue?
+  /// \note As messages can be added and removed by several different threads,
+  /// expect that the return value will change between two calls.
+  /// \return true if there are one or more messages in the queue, otherwise
+  /// false.
   virtual bool MessagesQueued();
```

Comment:
This has not changed in the commits, but that name does not suggest me this functions returns a boolean value, but the number of messages or the messages themselves instead.

Reply:
Will probably be done.